OpenWrt-specific:

Option to also set the hardware clock each time the system clock is
changed (-H).  For now, this just executes a hardwired /sbin/hwclock -w -u
(the openwrt default).

Index: ntpclient-2015/ntpclient.c
===================================================================
--- ntpclient-2015.orig/ntpclient.c	2016-01-28 14:33:32.501752349 -0800
+++ ntpclient-2015/ntpclient.c	2016-01-28 14:50:14.504082951 -0800
@@ -119,6 +119,7 @@
 	u32 time_of_send[2];
 	int live;
 	int set_clock;   /* non-zero presumably needs root privs */
+	int set_hw_clock;   /* non-zero needs root privs; calls hwclock -w -u */
 	int probe_count;
 	int cycle_time;
 	int goodness;
@@ -399,6 +400,9 @@
 	/* XXX should I do this if debug flag is set? */
 	if (ntpc->set_clock) { /* you'd better be root, or ntpclient will exit here! */
 		set_time(&xmttime);
+		if (ntpc->set_hw_clock) {
+			system("/sbin/hwclock -w -u");
+		}
 	}
 
 	/* Not the ideal order for printing, but we want to be sure
@@ -608,7 +612,7 @@
 	ntpc.cross_check=1;
 
 	for (;;) {
-		c = getopt( argc, argv, "c:" DEBUG_OPTION "f:g:h:i:lp:q:" REPLAY_OPTION "stD");
+		c = getopt( argc, argv, "c:" DEBUG_OPTION "f:g:h:i:lp:q:" REPLAY_OPTION "stDH");
 		if (c == EOF) break;
 		switch (c) {
 			case 'c':
@@ -661,6 +665,10 @@
 				daemon(0, 0);
 				break;
 
+			case 'H':
+				ntpc.set_hw_clock = 1;
+				break;
+
 			default:
 				usage(argv[0]);
 				exit(1);
